prow/config/config.go

```diff
@@ -499,16 +508,123 @@ type Plank struct {
 	JobURLPrefixDisableAppendStorageProvider bool `json:"jobURLPrefixDisableAppendStorageProvider,omitempty"`
 }
 
-func (p Plank) GetDefaultDecorationConfigs(repo string) *prowapi.DecorationConfig {
-	def := p.DefaultDecorationConfigs["*"]
-	if dcByRepo, ok := p.DefaultDecorationConfigs[repo]; ok {
-		return dcByRepo.ApplyDefault(def)
+type DefaultDecorationConfigEntry struct {
+	// Matching/filtering fields
+
+	// Repo matches against the "org/repo" that the presubmit or postsubmit is
+	// associated with. If the job is a periodic, extra_refs[0] is used. If the
+	// job is a periodic without extra_refs, the empty string will be used.
+	Repo *regexp.Regexp `json:"-"`
```

Comment:
Is there a particular reason must use regex instead of existing `*` style? Ask user to learn something new is fine imo as long as it's well communicated. I'm interested in the reasoning behind this change.

Reply:
I don't think there is an existing `*` (glob? wildcard?) matching style. The global config used key `*`, but that was just a special case literal match.
I think regexp matching is overkill here, but I figured it would be most appropriate for consistency with other Prow config fields. `run_if_changed`, `branches`, `exclude_branches`, `trigger`, etc. all use regular expressions rather than `*` matching. 

I also considered naming this field `OrgRepo` and just checking for string equality with the org or org/repo. This is consistent with other parts of Prow config, but I thought it more important that all the matching/filtering fields in this struct use a consistent matching system. I think it could be confusing (especially if we add more filter fields) if some things are matching literally (possibly against multiple fields), some are using `*` matching, and some are using regexp.
WDYT?

Reply:
> And for cluster filed, I think the only reason using regex instead of literal is because of global default right?

I was thinking it might be nice to reduce config size by using the cluster names to apply config across common clusters. e.g. All clusters with `-trusted` suffix or all clusters containing `istio`. This isn't that important though.

/assign @alvaroaleman 
WDYT Alvaro?

Reply:
Ok that all makes good sense to me.
What are your thoughts on the cluster field then? I had hoped to use a consistent matching system, but the more I think about it, the more I'm thinking we should just have `cluster` be a literal string match, at least initially. That is simpler and likely sufficient for existing use cases. With this new config slice pattern we can easily add more matching fields like `repo_re` or `cluster_re` down the road if we find a need for them.

Reply:
I don't really have an opinion regarding regex or not for `cluster`. It makes IMHO more sense than using it for `repo`, because for clusters you probably could have some kind of naming scheme that's based on trust domain but at the end of the day it boils down to what you need there and you know that better than me

Reply:
Ok I'm going to go with literal matching then, I think that will be sufficient for now. I like the idea of expanding functionality with something like `cluster_re` if we find that we need alternative matching systems for any of these fields later on.